### DISCUSSION TITLE
More lenient behavior regarding references

### DISCUSSION THREAD

**Discussion by memeplex** [typst/typst#3524] [General]
After reading https://github.com/typst/typst/issues/2930 I was wondering whether an option to support a more lenient compiler behavior regarding broken references may be desirable in order to support compiling only parts of a project. Even if the compiler is fast sometimes previewing a huge file is neither fast nor convenient. There are workarounds that I find acceptable, like [this one](https://github.com/typst/typst/issues/2930#issuecomment-1968384807) I proposed, but still having to "instrument" all references beforehand doesn't feel quite right.

I would like to know what do you think about this, if it is the result of a principled decision or just like that for the time being.

**Comment by laurmaedje** [comment]
I can imagine adding some way to make this more lenient down the road.
